src/volume.ts

```diff
@@ -656,4 +682,98 @@ export enum HostPathVolumeType {
    * A block device must exist at the given path.
    */
   BLOCK_DEVICE = 'BlockDevice'
-}
\ No newline at end of file
+}
+
+/**
+ * Options for an Nfs-based volume.
+ */
+ export interface NfsVolumeOptions {
+  /**
+   * path that is exported by the NFS server. More info: https://kubernetes.io/docs/concepts/storage/volumes#nfs
+   */
+  readonly path: string;
+
+  /**
+   * readOnly here will force the NFS export to be mounted with read-only permissions. Defaults to false. More info: https://kubernetes.io/docs/concepts/storage/volumes#nfs
+   *
+   * @default false. More info: https://kubernetes.io/docs/concepts/storage/volumes#nfs
+   */
+  readonly readOnly?: boolean;
+
+  /**
+   * server is the hostname or IP address of the NFS server. More info: https://kubernetes.io/docs/concepts/storage/volumes#nfs
+   */
+  readonly server: string;
+}
+
+export interface IscsiVolumeSourceSecretRef {
+  /**
+   * Name of the referred secret.
+   */
+  readonly name: string;
+}
+
+/**
+ * Represents an ISCSI disk. ISCSI volumes can only be mounted as read/write once. ISCSI volumes support ownership management and SELinux relabeling.
+ */
+ export interface IscsiVolumeOptions {
+  /**
+   * chapAuthDiscovery defines whether support iSCSI Discovery CHAP authentication
+   */
+  readonly chapAuthDiscovery?: boolean;
+
+  /**
+   * chapAuthSession defines whether support iSCSI Session CHAP authentication
+   */
+  readonly chapAuthSession?: boolean;
+
+  /**
+   * fsType is the filesystem type of the volume that you want to mount. Tip: Ensure that the filesystem type is supported by the host operating system. Examples: "ext4", "xfs", "ntfs". Implicitly inferred to be "ext4" if unspecified. More info: https://kubernetes.io/docs/concepts/storage/volumes#iscsi
+   */
+  readonly fsType?: string;
+
+  /**
+   * initiatorName is the custom iSCSI Initiator Name. If initiatorName is specified with iscsiInterface simultaneously, new iSCSI interface <target portal>:<volume name> will be created for the connection.
+   */
+  readonly initiatorName?: string;
+
+  /**
+   * iqn is the target iSCSI Qualified Name.
+   */
+  readonly iqn: string;
+
+  /**
+   * iscsiInterface is the interface Name that uses an iSCSI transport. Defaults to 'default' (tcp).
+   *
+   * @default 'default' (tcp).
+   */
+  readonly iscsiInterface?: string;
+
+  /**
+   * lun represents iSCSI Target Lun number.
+   */
+  readonly lun: number;
+
+  /**
+   * portals is the iSCSI Target Portal List. The portal is either an IP or ip_addr:port if the port is other than default (typically TCP ports 860 and 3260).
+   */
+  readonly portals?: string[];
+
+  /**
+   * readOnly here will force the ReadOnly setting in VolumeMounts. Defaults to false.
+   *
+   * @default false.
+   */
+  readonly readOnly?: boolean;
+
+  /**
+   * secretRef is the CHAP Secret for iSCSI target and initiator authentication
+   */
+  readonly secretRef?: IscsiVolumeSourceSecretRef;
```

Comment:
Secrets are referenced different with cdk8s-plus, by using the [`Secret.fromSecretName`](https://github.com/cdk8s-team/cdk8s-plus/blob/k8s-25/main/src/secret.ts#L103) method.

```suggestion
  readonly secretRef?: ISecret;
```

This also allows managed secrets to be passed here (i.e ones that were created using `new Secret(...)`)